res/controllers/Numark-Mixtrack-2-scripts.js

```diff
@@ -158,6 +159,12 @@ NumarkMixTrackII.selectKnob = function(channel, control, value, status, group) {
     }
 }
 
+NumarkMixTrackII.toggleSelectedSidebarItem = function(channel, control, value, status, group) {
+    if (value) {
+        engine.setValue("[Playlist]", "ToggleSelectedSidebarItem", 1);
```

Comment:
The `[Playlist]` controls are deprecated, but they still work
It's recommende to use the `[Library]` instead.

* Pressing the knob would be mapped to`[Library],GoToItem`, which triggers context-specific actions
https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[Library]-GoToItem
* turning the knob (`NumarkMixTrackII.selectKnob`) could use `[Library],Movevertical`
https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[Library]-MoveVertical

IIUC the Back button should move focus between the library widgets (search, sidebar, tracks).

Reply:
Thanks for letting me know. I might be able to look into this at some point in the future, but not right away.

Reply:
Hmm, I see with the propsed changes the 'directory mode' won't work as before. Now it would trigger move focus (like Tab key) and call `updateDirectoryAndFileLeds` which uses `directory_mode`.

Nowadays we have [`[Library],focused_widget`](https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[Library]-focused_widget)
I.e.  `directory_mode` is not required anymore and `updateDirectoryAndFileLeds` could use `[Library],focused_widget`: FOLDER whould be lit if it's `2` and FILE if it's `3`.

Are yo motivated to do that refactoring?

Reply:
I might like to do that at some time in the future but I can't promise. Maybe we can get this simple fix in so that controller based navigation works again? Without this fix the mouse or keyboard has to be used to go into a directory/folder when browsing the library.

Reply:
Oka, given my experience that "might like to do that at some time in the future" likely never happens due to shifted prios etc. (myself included ; ) and the fact that it'll take some weeks or months until 2.4.2 with the fixed mapping will be released (shortly before 2.5 I guess), we might as well try to fix it now.

If you don't have time/motivation to fix it yourself (potentially with another round of review) I may provide a commit this week, but I need you to test it (I don't have access to a Mixtrack Pro II).

Reply:
Well, I just bought a Pioneer DDJ-FLX4 so I won't be creating the patch for the Mixtrack. But, as long as I have not given back the Mixtrack to my friend, I can test a patch if you have it.